ingestion/src/metadata/ingestion/source/deltalake.py

```diff
@@ -0,0 +1,188 @@
+from metadata.ingestion.source.sql_source import SQLSourceStatus
+from metadata.utils.helpers import get_database_service_or_create
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+class DeltaLakeSourceConfig(ConfigModel):
+    database: str = "delta"
+    platform_name: str = "deltalake"
+    schema_filter_pattern: IncludeFilterPattern = IncludeFilterPattern.allow_all()
+    table_filter_pattern: IncludeFilterPattern = IncludeFilterPattern.allow_all()
+    service_name: str
+    service_type: str
+
+    def get_service_name(self) -> str:
+        return self.service_name
+
+
+class DeltaLakeSource(Source):
+    spark: SparkSession = None
+
+    def __init__(
+        self,
+        config: DeltaLakeSourceConfig,
+        metadata_config: MetadataServerConfig,
+        ctx: WorkflowContext,
+    ):
+        super().__init__(ctx)
+        self.config = config
+        self.metadata_config = metadata_config
+        self.service = get_database_service_or_create(config, metadata_config)
+        self.status = SQLSourceStatus()
+        # spark session needs to initiated outside the workflow and pass it through WorkflowContext
+        self.spark = ctx.spark
+
+    @classmethod
+    def create(
+        cls, config_dict: dict, metadata_config_dict: dict, ctx: WorkflowContext
+    ):
+        config = DeltaLakeSourceConfig.parse_obj(config_dict)
+        metadata_config = MetadataServerConfig.parse_obj(metadata_config_dict)
+        return cls(config, metadata_config, ctx)
+
+    def next_record(self) -> Iterable[OMetaDatabaseAndTable]:
+        schemas = self.spark.catalog.listDatabases()
+        for schema in schemas:
+            if not self.config.schema_filter_pattern.included(schema):
+                self.status.filter(schema, "Schema pattern not allowed")
+                continue
+            yield from self.fetch_tables(schema)
+
+    def fetch_tables(self, schema: str) -> Iterable[OMetaDatabaseAndTable]:
+        for table in self.spark.catalog.listTables(schema):
+            try:
+                database = table.database
+                table_name = table.name
+                if not self.config.table_filter_pattern.included(table_name):
+                    self.status.filter(
+                        "{}.{}".format(self.config.get_service_name(), table_name),
+                        "Table pattern not allowed",
+                    )
+                    continue
+                self.status.scanned(
+                    "{}.{}".format(self.config.get_service_name(), table_name)
+                )
+                table_columns = self._fetch_columns(schema, table_name)
+                fqn = f"{self.config.service_name}.{self.config.database}.{schema}.{table_name}"
+                if table.tableType and table.tableType.lower() != "view":
+                    table_description = self._fetch_table_description(table_name)
+                    table = Table(
```

Comment:
Noticed that we are using `table` as variable name for the for loop for listing tables above and also `table` on this line here to initialize the entity.

Should we disambiguate here (to avoid future issues), especially since python is dynamically typed?